### ISSUE TITLE
Replacing an explicit declaration with an equivalent compiler-generated should not trigger RS0017

### ISSUE THREAD

**Issue by VladimirReshetnikov** [dotnet/roslyn#1251]
See, for example, a declaration of an empty protected instance constructor in the class [`Microsoft.CodeAnalysis.Location`](http://source.roslyn.io/#Microsoft.CodeAnalysis/Diagnostic/Location.cs,15):
```cs
    public abstract class Location
    {
        protected Location()
        {
        }

        // ...
```

This declaration is redundant. If you remove it from source, the compiler generates exactly the same by default. So, it does not affect a shape of public API in any way.

But when I tried to remove it, the build failed with an error:
```
error RS0017: Symbol 'Microsoft.CodeAnalysis.Location.Location()' is part of the declared API, but is either not public or could not be found
```

__EXPECTED:__ This error does not occur.

**Comment by srivatsn** [comment]
While removing the constructor is not a breaking change, adding a parameter would be a breaking change. The analyzer simply points out situations which need to thought about to evaluate if it is a breaking change. It's not worth special-casing this case in the analyzer.

**Comment by VladimirReshetnikov** [comment]
@srivatsn I disagree and kindly recommend to reconsider this desicion. I do not only claim that removal of this constructor declaration is not a breaking change, I do claim that it is not a change at all. It does not change anything in the emitted assembly and is equivalent to re-formatting a code. Forcing a user to update any files to successfully build after this serves no useful purpose, and only places an unnecessary burden on the user.

**Comment by srivatsn** [comment]
I don't disagree. I was just saying that it's not an important enough case. I'll reopen the issue and mark it up for grabs. Feel free to submit a PR.

**Comment by Youssef1313** [comment]
@mavasani Please move to roslyn-analyzers to either fix it or add a test confirming if it was already fixed..
